src/Extension/TranslationTrait.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Bolt\Extension;
+
+use Bolt\Filesystem\Handler\DirectoryInterface;
+use Pimple as Container;
+
+/**
+ * Twig function/filter addition and interface functions for an extension.
+ *
+ * @author Carson Full <carsonfull@gmail.com>
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+trait TranslationTrait
+{
+
+    /**
+     * Call this in register method.
+     *
+     * @internal
+     */
+    final protected function addTranslations()
+    {
+        $app = $this->getContainer();
+
+        $translationDirectory = $this->getBaseDirectory()->getDir('translations');
+        if ($translationDirectory->exists()) {
+            foreach ($translationDirectory->getContents(true) as $fileInfo) {
+                $filename = explode('.', $fileInfo->getFilename());
+                if ($fileInfo->isFile()) {
+                    $translation = isset($filename[0]) ? $filename[0] : '';
+
+                    $app['translator']->addResource('yml', $app['paths']['extensionspath'] . DIRECTORY_SEPARATOR . $fileInfo->getPath(), $translation);
```

Comment:
First one… 

`$app['paths']` is Bolt v1 and we were going to remove in 3.0 but ran out of time.

From v2 onwards you should be using either `$app['filesystem']` (recommended for core stuff) or `$app['resources']` until the former takes over. e.g.
```
$app['resources']->getPath('extensions/ . $fileInfo->getPath());
```